### PR TITLE
s3 upload support in wysiwyg html

### PR DESCRIPTION
**Pull request by jrit**
@projectweekend can you take a look at this? Keeping to their coding standards FYI.

### CODE REVIEW

File: admin/api/s3.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+				path = s3Config.s3path ? s3Config.s3path + '/' : '';
+				originalname = file.originalname,
+				filetype = file.mimetype || file.type;
+
+			var s3Client = knox.createClient(s3Config);
+
+			s3Client.putFile(file.path, path + file.name, function(err, s3Response) {
+				var sendResult = function () {
+					if(err){
+						return res.send({ error: { message: err.message } });
```
Comment (projectweekend) [inline admin/api/s3.js]: Does this use Express middleware? If so you can just pass `err` to `next()`.

Comment (projectweekend) [inline admin/api/s3.js]: Maybe that won't work here because of something specific in Keystone. Just tossing it out there

Comment (jrit) [inline admin/api/s3.js]: It is an XHR thing here and should follow what is being done in https://github.com/vokal/keystone/blob/master/admin/api/cloudinary.js

Comment (projectweekend) [inline admin/api/s3.js]: Cool.

**Comment by projectweekend** [comment]
LGTM

**Pull request merged**